src/Raven.Server/Documents/SubscriptionStorage.cs

```diff
@@ -333,21 +278,44 @@ private unsafe DynamicJsonValue ExtractSubscriptionConfigValue(TableValueReader
             var timeOfLastClientActivity =
                 *(long*)tvr.Read(Schema.SubscriptionTable.TimeOfLastActivityIndex, out size);
             var criteria = new BlittableJsonReaderObject(tvr.Read(Schema.SubscriptionTable.CriteriaIndex, out size), size, context);
-
-            var criteriaInstance = JsonDeserialization.SubscriptionCriteria(criteria);
-            criteria.Dispose();
-
+            
             return new DynamicJsonValue
                 {
                     ["SubscriptionId"] = subscriptionId,
                     ["Criteria"] = criteria,
                     ["AckEtag"] = ackEtag,
-                    ["TimeOfSendingLastBatch"] = timeOfSendingLastBatch,
-                    ["TimeOfLastClientActivity"] = timeOfLastClientActivity,
+                    ["TimeOfSendingLastBatch"] = new DateTime(timeOfSendingLastBatch).ToString(CultureInfo.InvariantCulture),
+                    ["TimeOfLastClientActivity"] = new DateTime(timeOfLastClientActivity).ToString(CultureInfo.InvariantCulture),
                 };
         }
 
-        public void WriteSubscriptionTableValues(BlittableJsonTextWriter writer,
+        private void WriteSubscriptionStateData(SubscriptionState subscriptionState, DynamicJsonValue subscriptionData, bool writeHistory = false)
```

Comment:
You aren't actually writing here, you just add to the `subscriptionData`, the method name is misleading